keps/sig-node/4815-dra-partitionable-devices/README.md

```diff
@@ -0,0 +1,3546 @@
+   full devices and their paritions are represented as instances of this new
+   `PartitionableDevice` type and are listed right next to one another in the
+   top-level `Devices` list of a `ResourceSlice`.
+
+1. The `Includes` field serves to reference a set of "mixins" that a
+   `PartitionableDevice` can reference to extend the set of attributes and
+   capacities it defines explicitly. The goal being to reduce the overall
+   footprint of each device when a common set of attributes and capacities can
+   be applied. The mixins themselves are introduced as a list of top-level
+   objects directly next to the list of `Devices` inside a `ResourceSlice`.
+   They are not allocatable on their own.
+
+1. The `ConsumesCapacityFrom` field contains a list of *other* devices where the
+   capacity of the current device should be consumed if the scheduler decides
+   to allocate it. This essentially removes that capacity from any referenced
+   devices, rendering them unallocatable on their own.
+
+With these additions in place, the scheduler has everything it needs to support
+the dynamic allocation of both full devices and their (possibly overlapping)
+fixed-size partitions. That is to say, the scheduler now has the ability to
+"flatten" all devices by applying any mixins from their `Includes` fields as
+well as track any capacities consumed from one device by another through its
+`ConsumesCapacityFrom` field. More details on the actual algorithm the
+scheduler follows to make allocation decisions based on the
+`ConsumesCapacityFrom` field can be found in the Design Details section below.
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation. What is the desired outcome and how do we measure success?.
+The "Design Details" section below is for the real
+nitty-gritty.
+
+### User Stories ()
+
+Detail the things that people will be able to do if this KEP is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system. The goal here is to make this feel real for users without getting
+bogged down.
+
+#### Story 1
+
+#### Story 2
+
+### Notes/Constraints/Caveats (Optional)
+
+What are the caveats to the proposal?
+What are some important details that didn't come across above?
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+
+### Risks and Mitigations
+
+What are the risks of this proposal, and how do we mitigate? Think broadly.
+For example, consider both security and how this will impact the larger
+Kubernetes ecosystem.
+
+How will security be reviewed, and by whom?
+
+How will UX be reviewed, and by whom?
+
+Consider including folks who also work outside the SIG or subproject.
+-->
+
+## Design Details
```

Comment:
Any status updates?

Reply:
I just haven't had the time to fill this out properly yet.